### PR TITLE
Search screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 74517: Search Srceen](https://edu-redmine.sun-asterisk.vn/issues/74517)
## WHAT
- Show list in search screen
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HN_Android_2024_Image_Editor/assets/163242235/40d01b82-26e4-4f87-8bdf-b03211507160

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    interface View {
+        fun setupData(data: PhotoSearchModel?)
+
+        fun setupDataChangeQuery(data: PhotoSearchModel?)
+    }
+
+    interface Presenter : BasePresenter<View> {
+        fun searchPhotos(page: Int)
+
+        fun changeQuery(query: String)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt]: remove

File: app/src/main/java/com/example/imageEditor/model/Result.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.example.imageEditor.model
+
+import com.google.gson.annotations.SerializedName
+
+data class Result(
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/model/Result.kt]: Đặt tên cụ thể hơn đi em

Comment (daolq-2712) [reply app/src/main/java/com/example/imageEditor/model/Result.kt]: Có thể là ``Photo``

File: app/src/main/java/com/example/imageEditor/repository/SearchRepository.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import com.example.imageEditor.apiService.ApiImpl
+import com.example.imageEditor.apiService.NetworkService
+import com.example.imageEditor.base.OnListenProcess
+import com.example.imageEditor.model.PhotoSearchModel
+
+class SearchRepository(onListenProcess: OnListenProcess) {
+    private val networkService = NetworkService.getInstance(ApiImpl.getInstance(onListenProcess))
+
+    fun searchPhotos(
+        page: Int,
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/repository/SearchRepository.kt:12]:
```suggestion
        page: Int = 0,
```

Comment (daolq-2712) [reply app/src/main/java/com/example/imageEditor/repository/SearchRepository.kt:12]: set default values để khi click change query search không cần phải truyền page vào

File: app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.example.imageEditor.ui.search
+
+import com.example.imageEditor.base.BasePresenter
+import com.example.imageEditor.model.PhotoSearchModel
+
+interface SearchContract {
+    interface View {
+        fun setupData(data: PhotoSearchModel?)
+
+        fun setupDataChangeQuery(data: PhotoSearchModel?)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt]:
```suggestion
```
remove hàm này,
Nên clear currentList mỗi khi changeQuery

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

File: app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt (hunk 4 above)
Comment (ghost) [reply app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt]: cái listAdapter cứ summitList là nó clear currentList ạ, em tạo ra 2 cái để check  có add tiếp hay resetList ấy ạ

Comment (daolq-2712) [reply app/src/main/java/com/example/imageEditor/ui/search/SearchContract.kt]: thì mình resetList mỗi khi changeQuery 🥇 
@thold-3445

File: app/src/main/java/com/example/imageEditor/ui/search/SearchFragment.kt [hunk 5]
```diff
@@ -1,17 +1,133 @@
...
+            val layoutManager =
+                StaggeredGridLayoutManager(SPAN_COUNT, StaggeredGridLayoutManager.VERTICAL)
+            layoutManager.setGapStrategy(StaggeredGridLayoutManager.GAP_HANDLING_MOVE_ITEMS_BETWEEN_SPANS)
+            this.layoutManager = layoutManager
+            adapter = mAdapter
+        }
     }
 
     override fun initData() {
+        mPresenter.searchPhotos(mPageQuery)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/ui/search/SearchFragment.kt]:
```suggestion
        mPresenter.searchPhotos()
```

File: app/src/main/java/com/example/imageEditor/ui/search/SearchFragment.kt [hunk 6]
```diff
@@ -1,17 +1,134 @@
...
-class SearchFragment : BaseFragment<FragmentSearchBinding>() {
     override fun getViewBinding(inflater: LayoutInflater): FragmentSearchBinding {
         return FragmentSearchBinding.inflate(inflater)
     }
 
+    @SuppressLint("UseCompatLoadingForDrawables")
     override fun initView() {
+        mPresenter.setView(this)
+        binding?.radioGroup?.setOnCheckedChangeListener { group, checkedId ->
+            setBackgroundRadioButton(checkedId)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/ui/search/SearchFragment.kt:41]:
```suggestion
            setBackgroundRadioButton(checkedId)
            mAdapter.currentList.clear()
```

File: app/src/main/java/com/example/imageEditor/ui/search/SearchPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package com.example.imageEditor.ui.search
+
+import com.example.imageEditor.repository.SearchRepository
+import com.example.imageEditor.utils.ALL
+
+class SearchPresenter(private val searchRepository: SearchRepository) : SearchContract.Presenter {
+    private var query = ALL
```
Comment (daolq-2712) [inline app/src/main/java/com/example/imageEditor/ui/search/SearchPresenter.kt]:
```suggestion
    private var mQuery = ALL
```

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**